src/Store/StoreInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace SilverStripe\MFA\Store;
+
+use SilverStripe\Control\HTTPRequest;
+
+/**
+ * Represents a place for temporarily storing state related to a login or registration attempt.
+ */
+interface StoreInterface
+{
+    /**
+     * Create a store from the given request, using any initial state related to the request that has been persisted
+     *
+     * @param HTTPRequest $request
+     * @return StoreInterface
+     */
+    public static function create(HTTPRequest $request);
```

Comment:
We should be using Injectable for this. Not all StoreInterfaces could need the request (potentially) so if you put this in the SessionStore's constructor arguments it won't need to be part of the interface but can still be a requirement for construction, I don't know how injectable that will be though. You're already passing it as an argument to `save()` and `clear()`, so maybe the request argument could become stateless?

Reply:
Actually another thing - I don't know if we need static methods either. Once this instance is injectable you can do `Injector::inst()->get(StoreInterface::class)->clear()`

Reply:
Also v. difficult to test static methods

Reply:
It's more of a "named constructor" which makes it pretty reasonable to test. Injector is something I didn't really consider, nor did I consider the interface idea. We can't inject the request either so we'll need to inject this with a factory or declare the HTTPRequest in the interface.

Reply:
Yeah alright